### PR TITLE
*: add some necessary error check

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Some error can't be ignored

### What is changed and how it works?

Add necessary error check

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: syncer/sharding_group.go [hunk 1]
```diff
@@ -431,7 +431,10 @@ func (k *ShardingGroupKeeper) AddGroup(targetSchema, targetTable string, sourceI
 	if schemaGroup, ok := k.groups[schemaID]; !ok {
 		k.groups[schemaID] = NewShardingGroup(k.cfg.SourceID, k.shardMetaSchema, k.shardMetaTable, sourceIDs, meta, true)
 	} else {
-		schemaGroup.Merge(sourceIDs)
+		needShardingHandle, synced, remain, err = schemaGroup.Merge(sourceIDs)
```
Comment (WangXiangUSTC) [inline syncer/sharding_group.go]: is this a bug

Comment (csuzhangxc) [reply syncer/sharding_group.go]: Has any side effect with setting `needShardingHandle, synced, remain, err` here?

Comment (amyangfei) [reply syncer/sharding_group.go]: no side effect, but  `_, _, _, err = schemaGroup.Merge(sourceIDs)` seems better?

Comment (csuzhangxc) [reply syncer/sharding_group.go]: `_, _, _, err = schemaGroup.Merge(sourceIDs)` seems better

**Pull request merged**